kit/transport/http/api.go

```diff
@@ -165,20 +165,35 @@ func (a *API) Respond(w http.ResponseWriter, status int, v interface{}) {
 	}
 
 	w.Header().Set("Content-Type", "application/json; charset=utf-8")
-	enc := json.NewEncoder(writer)
+
+	// this marshal block is to catch failures before they hit the http writer.
```

Comment:
found this issue while testing. I had set an 0 value influxdb.ID type and it exploded in my face during encoding. However, my tests never caught it b/c a 200 was written during encoding. This fixes that up and also makes it viable for anyone else that may have hit this issue.

Reply:
did you mean 'body' instead of 'bod' here?

Reply:
good catch 👍, I did mean body